examples/snippets/servers/lowlevel/structured_output.py

```diff
@@ -0,0 +1,83 @@
+"""Low-level server example showing structured output support.
+
+This example demonstrates how to use the low-level server API to return
+structured data from tools, with automatic validation against output schemas.
+
+Run from the repository root:
+    uv run examples/snippets/servers/lowlevel/structured_output.py
+"""
+
+import asyncio
+from typing import Any
+
+import mcp.server.stdio
+import mcp.types as types
+from mcp.server.lowlevel import NotificationOptions, Server
+from mcp.server.models import InitializationOptions
+
+server = Server("example-server")
+
+
+@server.list_tools()
+async def list_tools() -> list[types.Tool]:
+    """List available tools with structured output schemas."""
+    return [
+        types.Tool(
+            name="calculate",
+            description="Perform mathematical calculations",
+            inputSchema={
+                "type": "object",
+                "properties": {"expression": {"type": "string", "description": "Math expression"}},
+                "required": ["expression"],
+            },
+            outputSchema={
+                "type": "object",
+                "properties": {
+                    "result": {"type": "number"},
+                    "expression": {"type": "string"},
+                },
+                "required": ["result", "expression"],
+            },
+        )
+    ]
+
+
+@server.call_tool()
+async def call_tool(name: str, arguments: dict[str, Any]) -> dict[str, Any]:
+    """Handle tool calls with structured output."""
+    if name == "calculate":
+        expression = arguments["expression"]
+        try:
+            # WARNING: eval() is dangerous! Use a safe math parser in production
+            result = eval(expression)
+            structured = {"result": result, "expression": expression}
+
+            # low-level server will validate structured output against the tool's
+            # output schema, and automatically serialize it into a TextContent block
+            # for backwards compatibility with pre-2025-06-18 clients.
```

Comment:
meganit: I'd maybe switch "automatically" to "additionally" to make it clearer that serialization is in addition to returning structuredContent?